In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
data = {
    "Hours Studies": [2, 3, 4, 5, 6, 7],
    "Hours Watching TV": [4, 3, 2, 1, 0, 0],
    "Outdoor Activity Time": [2, 4, 6, 8, 10, 12],
    "Hours Listening to Music": [2, 3, 4, 1, 5, 0],
    "Water Consumed": [5, 6, 5, 6, 4, 5],
    "Test Score": [65, 70, 75, 80, 85, 90]
}
df = pd.DataFrame(data)
df

Hours Studies  Hours Watching TV  Outdoor Activity Time  \
0              2                  4                      2   
1              3                  3                      4   
2              4                  2                      6   
3              5                  1                      8   
4              6                  0                     10   
5              7                  0                     12   

   Hours Listening to Music  Water Consumed  Test Score  
0                         2               5          65  
1                         3               6          70  
2                         4               5          75  
3                         1               6          80  
4                         5               4          85  
5                         0               5          90

In [3]:
def correlation_coeff(x, y):
    x_mean, y_mean = np.mean(x), np.mean(y)
    x_std, y_std = np.std(x), np.std(y)
    n = len(x)
    return np.sum((x-x_mean)*(y-y_mean)) / (n*x_std*y_std)

In [16]:
n =len(df.columns)
corr_test = []
for i in range(n-1):
    corr_test.append(correlation_coeff(df[df.columns[i]],df[df.columns[n-1]]))

corr_test = pd.DataFrame(corr_test,columns=['Test Score'],index=df.columns[:-1])
corr_test


Test Score
Hours Studies               1.000000
Hours Watching TV          -0.981981
Outdoor Activity Time       1.000000
Hours Listening to Music   -0.200000
Water Consumed             -0.355036

In [24]:
pos,neg,non = [],[],[]
for i in range(len(corr_test)):
    value = round(corr_test.iloc[i,0],2)
    if value > 0.5:
        pos.append(f"{corr_test.index[i]} - {corr_test.columns[0]} : {value}")                                
    elif value < -0.5:
        neg.append(f"{corr_test.index[i]} - {corr_test.columns[0]} : {value}")
    else:
        non.append(f"{corr_test.index[i]} - {corr_test.columns[0]} : {value}")
print("Positive Correlation")
for i in pos:
    print(i)
print("\nNegative Correlation")
for i in neg:
    print(i)
print("\nNon Correlation")
for i in non:
    print(i)

Positive Correlation
Hours Studies - Test Score : 1.0
Outdoor Activity Time - Test Score : 1.0

Negative Correlation
Hours Watching TV - Test Score : -0.98

Non Correlation
Hours Listening to Music - Test Score : -0.2
Water Consumed - Test Score : -0.36


## Positive Correlations

Hours Studies(+1): The direct link between studying and developing a greater comprehension of the subject matter can be used to explain the positive association between the amount of time spent studying and exam scores. Spending more time studying helps people learn topics thoroughly, practise problem-solving, and become more familiar with the content, all of which improve test performance.

Outdoor Activity(+1): The link between participating in outdoor activities and test scores is favourable, which shows that engaging in outdoor activities may help students do better in school. Exercise is proven to have positive effects on cognitive performance, emotional regulation, and general wellbeing, all of which can improve focus and academic results.


## Negative Correlation

Hours Watching TV(-0.98): The significant negative relationship between the amount of time spent watching TV and test scores highlights the potential time and attention losses associated with excessive television use when compared to studying and other worthwhile activities. This emphasises how crucial it is to prioritise learning and studying while effectively managing your time and avoiding distractions.


## No Significance Correlations

Hours listening to music (-0.2): The moderately favourable connection between the time spent listening to music and test scores (while not statistically significant) suggests that music may have a small positive influence on focus and relaxation during study sessions. Finding a balance that suits one's learning preferences and concentrate requirements is crucial, but it's important to note that individual preferences for studying settings might differ greatly.

Water Consumed (0.36): The moderately favourable association between the amount of water consumed and test results (while not statistically significant) raises the possibility that maintaining sufficient hydration may have a beneficial effect on cognitive performance. It's crucial to stay hydrated to prevent focus problems, weariness, and a loss of mental alertness, all of which can negatively impact exam performance.

These correlations provide insights into how various factors can influence academic performance, highlighting the significance of effective study habits, mindful time management, and overall well-being.